src/bootstrap/src/core/download.rs

```diff
@@ -21,6 +21,27 @@ fn try_run(config: &Config, cmd: &mut Command) -> Result<(), ()> {
     config.try_run(cmd)
 }
 
+fn extract_curl_version(out: &[u8]) -> (u16, u16) {
+    let out = &out[5..];
```

Comment:
Let's replace this with something like this:
```rust
let out = String::from_utf8_lossy(out);
// The output should look like this: "curl <major>.<minor>.<patch> ..."
out.lines().next().and_then(|line| line.split(" ").skip(1).next())
    .and_then(|version| semver::Version::parse(version).ok())
    .unwrap_or(Version::new(1, 0, 0))
```
And then we can compare it like this:
```rust
if curl_version() > semver::Version(7, 70, 0)
```

Reply:
> `if curl_version() > semver::Version(7, 70, 0)`

adding the patch level actually makes this incorrect, as it will return true for 7.70.1

fix is simply to use `>=` and the exact version it was added tho.

Reply:
Good point! :)